### ISSUE TITLE
[apex]  EmptyStatementBlock reports empty private constructor

### ISSUE THREAD

**Issue by robertStrunk** [pmd/pmd#4102]
**Affects PMD Version:** Apex PMD v0.5.9 -> PMD 6.42.0

**Rule:** [EmptyStatementBlock](https://pmd.github.io/latest/pmd_rules_apex_errorprone.html#emptystatementblock)

**Description:**
When you flag a constructor as private in order to prevent external instantiation, this rule is tripped if the constructor is empty. 

**Code Sample demonstrating the issue:**

```java
public with sharing class MySingletonClass {

    private MySingletonClass(){}
}
```

**Expected outcome:**
Constructor is not flagged -> false-positive

**Running PMD through:** [VSCode Extension](https://marketplace.visualstudio.com/items?itemName=chuckjonas.apex-pmd)

**Comment by adangel** [comment]
@robertStrunk Note that this rule has a property `reportEmptyPrivateNoArgConstructor` - by default it reports empty private no arg constructors, but you can disable it.

**Comment by robertStrunk** [comment]
Was not aware of that, thank you! @adangel
